src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationUtilities.cs

```diff
@@ -127,5 +131,27 @@ internal static class SemanticClassificationUtilities
                 throw ExceptionUtilities.Unreachable;
             }
         }
+
+        private static void CombineSpans(List<ClassifiedSpan> spans, IEnumerable<ClassifiedSpan> newSpans)
+        {
+            var spanSet = SharedPools.Default<HashSet<ClassifiedSpan>>().Allocate();
+            try
+            {
+                spanSet.AddAll(spans);
+
+                foreach (var span in newSpans)
+                {
+                    if (!spanSet.Contains(span))
```

Comment:
you can do "if (spanSet.Add(span)) { spans.Add(span); }".

Note: It would be nice if our APIs would guarantee ordering of the spans they return.